### PR TITLE
Decoupling register user button from CA users table 

### PR DESCRIPTION
**Pull request by selvaprakash92**
#### Type of change

- New feature

### CODE REVIEW

File: packages/apollo/src/components/ItemContainer/ItemContainer.js [hunk 1]
```diff
@@ -466,7 +466,7 @@ class ItemContainer extends Component {
 							onClick={button.fn}
 							kind={button.kind || (i !== this.props.addItems.length - 1 ? 'secondary' : 'primary')}
 							title={!button.tooltip ? (button.title ? translate(button.title) : translate(button.text)) : ''}
-							disabled={this.props.disableAddItem || this.props.loading || button.disabled}
+							disabled={button.caUserLoading ? button.caUserLoading :(this.props.disableAddItem || this.props.loading || button.disabled)}
```
[the remark this answers is not in the document]
Comment (dshuffma-ibm) [reply packages/apollo/src/components/ItemContainer/ItemContainer.js:469]: on second thought, we should make this more generic, change the name of `caUserLoading` to `decoupleFromLoading` and then change this line to:
```js
disabled={button.decoupleFromLoading ? false :(this.props.disableAddItem || this.props.loading || button.disabled)}
```

**Review by dshuffma-ibm — CHANGES_REQUESTED** [review]
good idea, needs a tweak, see comment

File: packages/apollo/src/components/ItemContainer/ItemContainer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (selvaprakash92) [reply packages/apollo/src/components/ItemContainer/ItemContainer.js]: @dshuffma-ibm Updated the review comments. Please check once

**Review by dshuffma-ibm — CHANGES_REQUESTED** [review]
so close, `decoupleFromLoading` should be true now

File: packages/apollo/src/components/CADetails/CADetails.js [hunk 2]
```diff
@@ -240,13 +240,15 @@ export class CADetails extends Component {
 				fn: () => {
 					this.generateCertificate(null);
 				},
-				disabled: !ActionsHelper.canEditComponent(this.props.feature_flags)
+				disabled: !ActionsHelper.canEditComponent(this.props.feature_flags),
+				decoupleFromLoading: false
```
Comment (dshuffma-ibm) [inline packages/apollo/src/components/CADetails/CADetails.js:244]: here

File: packages/apollo/src/components/CADetails/CADetails.js [hunk 3]
```diff
@@ -240,13 +240,15 @@ export class CADetails extends Component {
...
 			});
 			buttons.push({
 				text: 'register_user',
 				fn: this.openAddUser,
 				icon: 'plus',
-				disabled: !ActionsHelper.canEditComponent(this.props.feature_flags)
+				disabled: !ActionsHelper.canEditComponent(this.props.feature_flags),
+				decoupleFromLoading: false
```
Comment (dshuffma-ibm) [inline packages/apollo/src/components/CADetails/CADetails.js:251]: and here

**Comment by selvaprakash92** [comment]
@dshuffma-ibm , updated the value to true on both the places

File: packages/apollo/src/components/ItemContainer/ItemContainer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dshuffma-ibm) [reply packages/apollo/src/components/ItemContainer/ItemContainer.js]: great

**Review by dshuffma-ibm — APPROVED** [review]
+1

**Pull request merged**